src/utils/validate-handle.ts

```diff
@@ -0,0 +1,51 @@
+import type { Indexer } from '../indexer'
+
+export type ValidateHandleErrorKind =
+  | 'existed'
+  | 'lengthInvalid'
+  | 'charsInvalid'
+
+export type ValidateHandleResult =
+  | { isValid: true; error?: undefined; errorMsg?: undefined }
+  | { isValid: false; error: ValidateHandleErrorKind; errorMsg: string }
+
+const ERROR_MSG_MAP: Record<ValidateHandleErrorKind, string> = {
+  existed: 'The handle is already in use.',
+  lengthInvalid: 'The handle must be between 3 and 31 characters.',
+  charsInvalid:
+    'The handle must only contain lower-case letters, numbers, hyphens (-), or underscores (_).',
+}
+
+export async function validateHandle(
+  handle: string,
+  indexer: Indexer,
```

Comment:
Make it optional. Defaults to createIndexer()

Reply:
Using `indexer: Indexer = createIndexer()` will create a new indexer every time `validateHandle` is called if the user forgets to pass in the indexer. Is it really necessary to tweak the code like that? 🤔️

Reply:
Or we can add a shared global instance `defaultIndexer`.

Reply:
How about we leave the indexer as optional, and simply skip `checkIfExisted` if the indexer does not exist?

Reply:
Good idea.

Reply:
Updated 896ca8f60566dde4abf3bc9eb7034354716be030

Reply:
> How about we leave the indexer as optional, and simply skip checkIfExisted if the indexer does not exist?

This approach doesn't seem intuitive to me, since the behaviour of this function should remain consistent regardless of the parameters. Otherwise, it would require a detailed jsdoc to explain this to developers.

Reply:
The primary use case for validateHandle involves minting a new character and updating the handle for an existing character. Both scenarios require checking if the new handle is already in use.

734f4661cb6837415cc8b1293c830d476272ccf4 marks the indexer as required and lets the caller pass the indexer instance, without providing a default indexer. In these situations, an indexer will be needed in other steps anyway.